sched/jobDefSerializer.go

```diff
@@ -0,0 +1,112 @@
+)
+
+func Deserialize(serializedVal []byte, serializeType string) (*JobDefinition, error) {
+
+	// create the thrift deserializer
+	//deserializer := thrift.NewTDeserializer();
+	deserializer, err := getDeserializer(serializeType)
+	if (err != nil) {
+		return nil, err
+	}
+
+	// create the thrift jobDef struct for the saga log
+	sagaLogJobDef := sagalog.NewJobDefinition()
+	// parse the byte string into the thrift jobDef struct
+	if err := deserializer.Read(sagaLogJobDef, serializedVal); err != nil {
+		return nil, err
+	}
+
+	// make a sched JobDefintion from the thrift JobDefinition for the saga log
+	var schedJobDef = JobDefinition{}
+	schedJobDef.JobType = sagaLogJobDef.JobType
+	schedJobDef.Tasks = make(map[string]TaskDefinition)
+	for taskName, sagaLogTaskDefTask := range sagaLogJobDef.Tasks {
+		argvs := (*(*sagaLogTaskDefTask).Command).Argv
+		envVars := (*(*sagaLogTaskDefTask).Command).EnvVars
+		timeout, _ := time.ParseDuration(fmt.Sprintf("%dns", (*(*sagaLogTaskDefTask).Command).Timeout))
+		snapshotId := (*(*sagaLogTaskDefTask).Command).SnapshotId
+		command := runner.NewCommand(argvs, envVars, timeout, snapshotId)
+		schedJobDef.Tasks[taskName] = TaskDefinition{*command}
+	}
+	return &schedJobDef, nil
+}
+
+func Serialize(jobDef *JobDefinition, serializeType string) ([]byte, error) {
+
+	if (jobDef == nil) {
+		return nil, fmt.Errorf("Cannot serialize a nil object")
+	}
+	serializer, err := getSerializer(serializeType)
+	if (err != nil) {
+		return nil, err
+	}
+
+	// allocate a thrift JobDefinition for saga log
+	sagalogJobDefinition := sagalog.NewJobDefinition()
+
+	// copy the sched JobDefinition properties to the thrift jobDefinition structure
+	sagalogJobDefinition.JobType = jobDef.JobType
+	sagalogJobDefinition.Tasks = make(map[string]*sagalog.TaskDefinition)
+	for taskName, taskDef := range jobDef.Tasks {
+		sagaTaskDef := sagalog.NewTaskDefinition()
+		(*sagaTaskDef).Command = sagalog.NewCommand()
+		(*(*sagaTaskDef).Command).Argv = taskDef.Argv
+		(*(*sagaTaskDef).Command).EnvVars = taskDef.EnvVars
+		(*(*sagaTaskDef).Command).SnapshotId = taskDef.SnapshotId
+		(*(*sagaTaskDef).Command).Timeout = taskDef.Timeout.Nanoseconds()
+		sagalogJobDefinition.Tasks[taskName] = sagaTaskDef
+	}
+
+	if serializedVal, err := serializer.Write(sagalogJobDefinition); err != nil {
+		return nil, err
+	} else {
+		return serializedVal, nil
+	}
+
+}
+
+func getSerializer(serializeType string) (*thrift.TSerializer, error) {
+
+	switch serializeType {
+	case JsonSerialize:
+		transport := thrift.NewTMemoryBufferLen(1024)
+		protocol := thrift.NewTJSONProtocol(transport)
+		serializer := thrift.TSerializer{transport, protocol}
+		return &serializer, nil
+	case BinarySerialize:
+		serializer := thrift.NewTSerializer()
+		return serializer, nil
+	default:
+		return nil, fmt.Errorf("Invalid serializer type: %s\n", serializeType)
+	}
+
+}
+
+func getDeserializer(serializeType string) (*thrift.TDeserializer, error) {
```

Comment:
You could just have vars for jsonDeserializer and jsonSerializer?

Go makes it super easy to make an interface and then just add a function to an otherwise-empty struct?